### DISCUSSION TITLE
Compilation errors in Visual Studio 2019 community

### DISCUSSION THREAD

**Discussion by threedslider** [o3de/o3de#3679] [Help Wanted] [answered]
I have compiled with Visual Studio 2019 Community for O3DE version 2107.1 from source code to target x64, it takes so long time and very well but until in nearly final VS 2019 is stopped et gets 13 errors as here in screenshot 

![screenshot](https://user-images.githubusercontent.com/16051147/131254641-bffcd094-0c3f-4917-b00c-bdc774c5c89d.jpg)

How to resolve it ? Thanks

**Comment by threedslider** [comment]
Ok, I will try to resolve from this link ==> [errors for resource files](https://github.com/o3de/o3de/discussions/1890)

I will say you if it works.

See you later

**Comment by threedslider** [comment]
Well after updating my Visual studio 2019 Community version 16.11.2, cloned it in lastest version from  the repo of O3DE on my local folder and sync with git lfs ... some problem is cmake, it says bin/CMakeFiles/generate.stamp is out-of-date, and :

1>CMake Error at CMakeLists.txt:17 (project):
1>  The CMAKE_C_COMPILER:
1>
1>    C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe
1>
1>  is not a full path to an existing compiler tool.

a bit later, it says too :

o3de/Code/Framework/AzCore/AzCore/Serialization/ObjectStream.cpp(1519,42): error C5233: la capture de lambda explicite 'classData' n'est pas utilisée
2>            auto closeElementCB = [this, classData]()
2>                                                        ^

Thus AzCore.lib is failed to compile ...

Sorry, before version 2107.1 of O3DE was working well except to resource files in final compilation, now here is the beginning it gets erros :/

How to resolve it ?

**Comment by AlexNewtown** [comment]
these files are icon files，you can try to copy some other icon files and rename to these files. You can try this as a temporary solution...

**Comment by amznestebanpapp** [comment, marked as answer]
Sorry for the late reply

When updating VS delete the generated build folder and generate again.
CMake doesnt regenerate the paths to the toolchain, so likely is looking for 
```
C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe`
```
But the new version of VS does not have that toolchain.
Deleting the build folder and generating again should pick the new toolchain

**Reply by amznestebanpapp** [reply]
@AlexNewtown let me know if that doesnt fix your issue, marking this as answered
